### PR TITLE
Publish cli

### PR DESCRIPTION
**Pull request by rylandg**
The only testing I can really do is the bad case (which I did), make without NPM_TAG.

### CODE REVIEW

**Comment by rylandg** [comment]
@michaeladda I edited the original PR note to say the testing. It wasn't tested extensively. Any recommendations on viable testing without actually publishing?

File: Makefile [hunk 1]
```diff
@@ -45,19 +45,26 @@ test: build
 			bash -c "cd /home/dockeruser/binaris && npm run test"
 
 .PHONY: publish
-publish: build require-npm-creds
+publish: build require-npm-creds require-npm-tag
+		version=$(shell cat package.json | jq -r ".version")
+		git tag $(version)
```
Comment (michaeladda) [inline Makefile]: shouldn't NPM_TAG be here too?
otherwise you assume single npm tag per version and the mapping is not 1 to 1

Comment (rylandg) [reply Makefile]: We've never discussed the mapping not being 1:1. We use `NPM_TAG` in the target you've commented on.

Comment (rylandg) [reply Makefile]: Oh I see you mean instead of the `version`. I guess that makes sense, this is just the way we do it elsewhere.

**Comment by rylandg** [comment]
So I've made some changes and tested this manually. I still have not tested it with a full end-to-end because `publish` doesn't work the way you described. Basically we will need a second job which does `npm dist-tag` so we can add tags to existing versions. 

I did manually verify the git portion works and the correct NPM tag propagate into the environment.

**Pull request merged**